hyperspy/tests/io/test_mrcz.py

```diff
@@ -177,6 +178,8 @@ def test_MRC(self, dtype, compressor, clevel, lazy):
         print("MRCZ test ({}, {}, {}, lazy:{}) finished in {} s".format(
             dtype, compressor, clevel, lazy, perf_counter() - t_start))
 
+    @pytest.mark.xfail(sys.platform.startswith("win"),
```

Comment:
Why use xfail to skip instead of skip/skipif? You don't _expect_ it to fail, rather, it just often does.

Reply:
I guess, change the description to not say "skip" - since this test will in fact run

Reply:
OK, in this case I was unsure which one to use and I changed it from `skipif` to `xfail`...

Maybe @robbmcleod would have an idea why this is happening significantly often (about 25%?)  since we moved from appveyor to azure pipeline, yesterday?

Reply:
@ericpre  is it the test that fails with this?

```
>       [version, versionlz, flags, typesize] = np.fromfile(filehandle, dtype='uint8', count=4)
E       ValueError: not enough values to unpack (expected 4, got 0)

C:\Miniconda\envs\test_env\lib\site-packages\mrcz\ioMRC.py:449: ValueError```

According to [`np.fromfile`](https://numpy.org/doc/stable/reference/generated/numpy.fromfile.html?highlight=fromfile#numpy.fromfile)

> Do not rely on the combination of tofile and fromfile for data storage, as the binary files generated are not platform independent. In particular, no byte-order or data-type information is saved. Data can be stored in the platform independent .npy format using save and load instead.

Reply:
Wait, what is this `do_async` test trying to achieve?

Reply:
```python
if do_async:
            # Poll file on disk since we don't return the
            # concurrent.futures.Future
            t_stop = perf_counter() + MAX_ASYNC_TIME
            sleep(0.005)
            while(perf_counter() < t_stop):
                try:
                    fh = open(mrcName, 'a')
                    fh.close()
                    break
                except IOError:
                    sleep(0.001)
            print("Time to save file: {} s".format(
                perf_counter() - (t_stop - MAX_ASYNC_TIME)))
            sleep(0.005)
```

Reply:
This seems like a really odd "test" anyway, unless I'm missing something.

Reply:
@ericpre  I would wager that probably changing some of those `sleep()` periods will make it "work" better on azure...

Reply:
Yes, this sounds very sensible, there is already some precedent on this: https://github.com/hyperspy/hyperspy/pull/1852.

I increased a bit one `sleep` call and so far 2 build runs fine, I will run some build by restarting manually on my azure: https://dev.azure.com/ericpre/hyperspy/_build?definitionId=10&_a=summary and see it is good enough.

Reply:
The asynchronous read/write tests are probably not required. I recall at the time you wanted 100 % test coverage, and `mrcz.asyncWriteMRC` doesn't return the future. If you wanted to do it properly without the `sleep` call, probably polling `psutil.Process().open_files()` would work. I don't recall if `psutil` is already a dependency or not.

Reply:
Thanks @robbmcleod for the clarification and the pointer!